tools/gmhazard_scripts/db_creation/empirical_db/calc_emp_flt.py

```diff
@@ -46,54 +40,132 @@ def calculate_flt(
         suffix=suffix,
     )
 
-    with sc.dbs.SiteSourceDB(site_source_db_ffp) as distance_store:
-        fault_df, n_stations, site_df, work = common.get_work(
-            distance_store, vs30_ffp, z_ffp, rank, size
-        )
+    with sc.dbs.SiteSourceDB(site_source_db_ffp, writeable=False) as distance_store:
+        distance_stations = np.asarray(distance_store.stored_stations())
 
-        rupture_df = fault_df.copy(deep=True)
-        rupture_df.columns = ["rupture_name"]
+        fault_df, _, site_df, work = common.get_work(
+            distance_store, vs30_ffp, z_ffp, None, None
+        )
 
-        for site_index, site in work.iterrows():
-            max_dist = common.get_max_dist_zfac_scaled(site)
-            if distance_store.has_station_data(site.name):
-                print(
-                    f"Processing site {(site_df.index.get_loc(site_index) + 1)} / {n_stations} - Rank: {rank}"
+    # Drop stations for which there is no distance data
+    site_df = site_df.loc[np.isin(site_df.index.values, distance_stations)]
+    n_stations = site_df.shape[0]
+
+    tect_type_model_dict = empirical_factory.read_model_dict(tect_type_model_dict_ffp)
+
+    rupture_df = fault_df.copy(deep=True)
+    rupture_df.columns = ["rupture_name"]
+
+    for ix in range(math.ceil(n_stations / 1000)):
+        cur_site_df = site_df.iloc[(ix * 1000) : (ix + 1) * 1000]
```

Comment:
Is this just making sure we only calculate for 1000 stations at a time? Is this for memory issues?

Reply:
Yeah just keeps the amount of data manageable (it might not actually be needed)